### PR TITLE
Quarterly release branch update and cherry-pick

### PR DESCRIPTION
**Pull request by garyschulte**
Signed-off-by: garyschulte <garyschulte@gmail.com>

## PR description
Quarterly release for 21.10.0.  In addition to the cumulative release candidates (RC1, RC2, RC3, RC4):

* cherry-pick merge of arrow-glacier
* update CHANGELOG.md for quarterly release

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: config/src/main/java/org/hyperledger/besu/config/GenesisConfigOptions.java [hunk 1]
```diff
@@ -71,8 +71,9 @@
 
   OptionalLong getLondonBlockNumber();
 
-  // TODO EIP-1559 change for the actual fork name when known
-  OptionalLong getAleutBlockNumber();
+  OptionalLong getArrowGlacierBlockNumber();
+
+  OptionalLong getBaseFeePerGas();
```
Comment (shemnon) [inline config/src/main/java/org/hyperledger/besu/config/GenesisConfigOptions.java:76]: Can all the BaseFeePerGas methods that are added be removed?  It's not needed for the bomb and not referenced anywhere else in this PR.

File: config/src/main/java/org/hyperledger/besu/config/JsonGenesisConfigOptions.java [hunk 2]
```diff
@@ -252,31 +252,25 @@ public OptionalLong getBerlinBlockNumber() {
...
+
+  @Override
+  public OptionalLong getArrowGlacierBlockNumber() {
+    return getOptionalLong("arrowglacierblock");
   }
 
   @Override
-  public OptionalLong getAleutBlockNumber() {
-    return getOptionalLong("aleutblock");
+  public OptionalLong getBaseFeePerGas() {
```
Comment (shemnon) [inline config/src/main/java/org/hyperledger/besu/config/JsonGenesisConfigOptions.java:264]: Not needed.

File: config/src/main/java/org/hyperledger/besu/config/StubGenesisConfigOptions.java [hunk 3]
```diff
@@ -37,9 +37,9 @@
   private OptionalLong muirGlacierBlockNumber = OptionalLong.empty();
   private OptionalLong berlinBlockNumber = OptionalLong.empty();
   private OptionalLong londonBlockNumber = OptionalLong.empty();
+  private OptionalLong arrowGlacierBlockNumber = OptionalLong.empty();
 
-  // TODO EIP-1559 change for the actual fork name when known
-  private final OptionalLong aleutBlockNumber = OptionalLong.empty();
+  private OptionalLong baseFeePerGas = OptionalLong.empty();
```
Comment (shemnon) [inline config/src/main/java/org/hyperledger/besu/config/StubGenesisConfigOptions.java:42]: Not Needed.

File: config/src/main/java/org/hyperledger/besu/config/StubGenesisConfigOptions.java [hunk 4]
```diff
@@ -180,18 +180,18 @@ public OptionalLong getLondonBlockNumber() {
...
   @Override
-  // TODO EIP-1559 change for the actual fork name when known
-  public OptionalLong getAleutBlockNumber() {
-    return aleutBlockNumber;
+  public OptionalLong getArrowGlacierBlockNumber() {
+    return arrowGlacierBlockNumber;
+  }
+
+  @Override
+  public OptionalLong getBaseFeePerGas() {
```
Comment (shemnon) [inline config/src/main/java/org/hyperledger/besu/config/StubGenesisConfigOptions.java:188]: Not Needed.

File: config/src/main/java/org/hyperledger/besu/config/StubGenesisConfigOptions.java [hunk 5]
```diff
@@ -415,6 +415,16 @@ public StubGenesisConfigOptions londonBlock(final long blockNumber) {
     return this;
   }
 
+  public StubGenesisConfigOptions arrowGlacierBlock(final long blockNumber) {
+    arrowGlacierBlockNumber = OptionalLong.of(blockNumber);
+    return this;
+  }
+
+  public StubGenesisConfigOptions baseFeePerGas(final long baseFeeOverride) {
```
Comment (shemnon) [inline config/src/main/java/org/hyperledger/besu/config/StubGenesisConfigOptions.java:423]: Not Needed.

File: ethereum/retesteth/src/main/java/org/hyperledger/besu/ethereum/retesteth/methods/TestSetChainParams.java [hunk 6]
```diff
@@ -122,6 +122,8 @@ private static String modifyGenesisFile(final String initialGenesis) {
     maybeMoveToNumber(params, "istanbulForkBlock", config, "istanbulBlock");
     maybeMoveToNumber(params, "muirGlacierForkBlock", config, "muirGlacierBlock");
     maybeMoveToNumber(params, "berlinForkBlock", config, "berlinBlock");
+    maybeMoveToNumber(params, "londonForkBlock", config, "londonBlock");
+    maybeMoveToNumber(params, "muirGlacierForkBlock", config, "muirGlacierBlock");
```
Comment (garyschulte) [inline ethereum/retesteth/src/main/java/org/hyperledger/besu/ethereum/retesteth/methods/TestSetChainParams.java:126]: this needs to be arrowGlacier.

**Review by shemnon — COMMENTED** [review]
Please remove the _added_ baseFeePerGas lines.  It's coming back in dot-one but this should be a minimum required change cherry pick.

File: config/src/main/java/org/hyperledger/besu/config/GenesisConfigOptions.java (hunk 1 above)
Comment (garyschulte) [reply config/src/main/java/org/hyperledger/besu/config/GenesisConfigOptions.java:76]: yup.  can do.  This is one of the artifacts of the merge strategy for the cherry-pick.

**Review by shemnon — APPROVED** [review]

**Pull request merged**